nodeJS/express/controllers.md

```diff
@@ -0,0 +1,459 @@
+### Introduction
+
+The controller’s job is really to act as the ultimate middleman. It knows which questions it wants to ask the model, but lets the model do all the heavy lifting for actually solving those questions. It knows which view it wants to render and send back to the browser, but lets the view itself take care of putting all that HTML together. That’s why it’s a “controller” - it's smart enough to know what to do, then it delegates all the hard work.
+
+A controller comes into play whenever a request hits the server and a route handler matches the requested route. The route handler determines which controller should handle the request based on the defined middleware chain. The appropriate controller then takes over and performs the necessary actions to fulfill the request. This could involve retrieving data from the model, processing the data, making decisions based on business logic, or updating the model with new data.
+
+Once the controller has completed its tasks, it passes the processed data to the view, which renders the data into a format suitable for sending back to the client. Typically, this would be HTML. Later, when we cover building APIs, we can also send JSON responses like with the APIs that you've previously encountered e.g. Giphy API.
+
+Ultimately, controllers are pretty much just functions with well-defined responsibilities as part of the MVC pattern.
```

Comment:
What do you think if we move this bit to # Controllers instead of the introduction, since I now realize that we are just re-iterating the same stuff

New intro would look like:

> The controller's job is really to act as the ultimate middleman. It knows which questions it wants to ask the model, but lets the model do all the heavy lifting for actually solving those questions. It knows which view it wants to render and send back to the browser, but lets the view itself take care of putting all that HTML together. That's why it's a "controller" - it's smart enough to know what to do, then it delegates all the hard work. 
> 
> Ultimately, the controller is the brains of the operation ensuring that each component plays its part in delivering the final result and is really just a function with well-defined responsibilities as part of the MVC pattern.

And for the # Controllers section (removed the sentence that I think is the same):

> As said earlier, controllers are just functions, they also classify as a middleware (at least in the express world) that are used by route handlers.
> 
> A controller comes into play whenever a request hits the server and a route handler matches the requested route. The route handler determines which controller should handle the request based on the defined middleware chain. The appropriate controller then takes over and performs the necessary actions to fulfill the request. This could involve retrieving data from the model, processing the data, making decisions based on business logic, or updating the model with new data.
> 
> Once the controller has completed its tasks, it passes the processed data to the view, which renders the data into a format suitable for sending back to the client. Typically, this would be HTML. Later, when we cover building APIs, we can also send JSON responses like with the APIs that you've previously encountered e.g. Giphy API.

Reply:
I like it!